tools/unicode_properties_parse/format_width_estimate_intervals.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# The following code generates data for _Width_estimate_intervals_v2 in <format>.
+
+import re
+from enum import Enum
+from typing import TextIO
+from pathlib import Path
+
+
+# Width estimation.
+class UnicodeWidth(Enum):
+    IS_1: int = 1
+    IS_2: int = 2
+
+
+class UnicodeWidthTable:
+    # A valid Unicode code point won't exceed MAX_CODE_POINT.
+    MAX_CODE_POINT: int = 0x10FFFF
+    TABLE_SIZE: int = MAX_CODE_POINT + 1
+
+    def __init__(self):
+        self.table = [UnicodeWidth.IS_1] * (self.TABLE_SIZE)
+
+    # "rng" denotes a right-closed range.
+    def fill_range(self, rng: tuple, width: int):
+        from_, to_ = rng
+        assert from_ <= to_, "invalid range"
+        assert to_ <= self.MAX_CODE_POINT, "invalid range"
+        self.table[from_ : to_ + 1] = [width] * (to_ - from_ + 1)
+
+    def width_estimate_intervals(self):
+        """
+        Divide [0..MAX_CODE_POINT] into ranges with different width estimations.
+        Represent each range with their starting values.
+        The starting value of the first range is always 0 and omitted.
+        The width estimation should be 1 for the first range, then alternate between 2 and 1.

```

Comment:
```suggestion
        Creates a string representation of the map (in `self.table`) from
        unicode code points to their width, using hexadecimal unsigned integer literals.
        Since there are long runs of code points of one width or the other,
        this representation is a list of code points where the width switches.
        Additionally, the width is assumed to start at `1` from the beginning of the list.
        For example, `[1, 1, 2, 2, 2, 1]` would be represented as `"0x2u, 0x5u"`.
```

How do you feel about this?

Reply:
Looks good to me.